crates/core/src/factories/core_fetch/test.rs

```diff
@@ -97,41 +99,71 @@ mod tests {
             }
         }
         #[derive(Debug)]
-        struct K;
-        impl KitsuneHandler for K {
+        struct MockKitsuneHandler;
+        impl KitsuneHandler for MockKitsuneHandler {
             fn create_space(
                 &self,
                 _space_id: SpaceId,
             ) -> BoxFut<'_, K2Result<DynSpaceHandler>> {
-                let s: DynSpaceHandler = Arc::new(S);
+                let s: DynSpaceHandler = Arc::new(MockSpaceHandler);
                 Box::pin(async move { Ok(s) })
             }
         }
-        k.register_handler(Arc::new(K)).await.unwrap();
-        let s = k
+        kitsune
+            .register_handler(Arc::new(MockKitsuneHandler))
+            .await
+            .unwrap();
+        let space = kitsune
             .space(kitsune2_test_utils::space::TEST_SPACE_ID)
             .await
             .unwrap();
         let local_agent: DynLocalAgent =
             Arc::new(kitsune2_test_utils::agent::TestLocalAgent::default());
-        s.local_agent_join(local_agent.clone()).await.unwrap();
+        space.local_agent_join(local_agent.clone()).await.unwrap();
 
-        let url = s.current_url().unwrap();
+        let url = space.current_url().unwrap();
 
         assert!(GOT_SPACE.load(std::sync::atomic::Ordering::SeqCst));
 
         #[derive(Debug)]
-        struct T;
-        impl TxBaseHandler for T {}
-        impl TxHandler for T {}
+        struct MockTxHandler;
+        impl TxBaseHandler for MockTxHandler {}
+        impl TxHandler for MockTxHandler {}
 
         let builder = Arc::new(crate::default_test_builder());
-        let tx = builder
+        let tx: Arc<dyn Transport> = builder
             .transport
-            .create(builder.clone(), Arc::new(T))
+            .create(builder.clone(), Arc::new(MockTxHandler))
             .await
             .unwrap();
 
+        // We need to add an agent info of the sending peer to the receiving
```

Comment:
This is the actual fix here. The message was getting dropped now that messages from peers where we don't have any agents in the peer store get dropped. So we need to add an agent info of the sending peer into the receiving peer's peer store.